scala/src/main/scala/smile/clustering/distances/Binary/HammingDistance.scala

```diff
@@ -0,0 +1,15 @@
+package smile.clustering.distances
+
+class HammingDistance extends Binary_Distance
+{
+	/**
+	  * The famous hamming distance implemented in its fast mono thread scala version
+	  */
+	override def distance(vector1: Array[Int], vector2: Array[Int], specialArg: String = "") : Double = 
```

Comment:
`specialArg` is not recommended. If a distance class need customization, it should be passed in the constructor. Please check our implementation in `smile.math.distance` for general guideline.